drivers/power/max42500/max42500.c

```diff
@@ -0,0 +1,904 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "max42500.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+#include "no_os_alloc.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+#define CRC8_PEC        0x07      /* Implements Polynomial X^8 + X^2 + X^1 +1 */
+#define	ECOMM           9         /* General communication error */
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/******************************************************************************/
+
+/**
+ * @brief 8-bit CRC computation
+ * @param data - Data buffer.
+ * @param len - Buffer length.
+ * @return 8-bit CRC value
+ */
+static uint8_t crc8(uint8_t *data, uint8_t len)
+{
+	uint8_t crc = 0;
+	bool msb_set;
+
+	for (uint8_t idx = 0; idx < len; idx++) {
+		crc ^= data[idx];
+
+		for (uint8_t loop=0; loop < 8; loop++) {
+			msb_set = ((crc & 0x80) == 0x80);
+			crc <<= 1;
+
+			/* Divide by PEC if bit 8 was set */
+			if (msb_set)
+				crc ^= CRC8_PEC;
+		}
+	}
+
+	return crc;
+}
+
+/**
+ * @brief 8-bit watchdog key computation
+ * @param curr_wd_key - Current watchdog key value.
+ * @return New 8-bit watchdog key value
+ */
+static uint8_t max42500_new_watchdog_key(uint8_t curr_wd_key)
+{
+	return curr_wd_key;
+}
+
+/**
+ * @brief Set device state through EN0 and EN1 pins
+ * @param state - Device state.
+ * @return 0 in case of success, error code otherwise
+ */
+int max42500_set_state(struct max42500_dev *desc, enum max42500_state state)
+{
+	if (!desc)
+		return -EINVAL;
+
+	int ret;
+	uint8_t en0;
+	uint8_t en1;
+
+	switch (state) {
+	case MAX42500_STATE_OFF:
+		en0 = NO_OS_GPIO_LOW;
+		en1 = NO_OS_GPIO_LOW;
+		break;
+
+	case MAX42500_STATE_SLEEP:
+		en0 = NO_OS_GPIO_HIGH;
+		en1 = NO_OS_GPIO_LOW;
+		break;
+
+	case MAX42500_STATE_ON:
+		en0 = NO_OS_GPIO_HIGH;
+		en1 = NO_OS_GPIO_HIGH;
+		break;
+
+	default:
+		en0 = NO_OS_GPIO_HIGH;
+		en1 = NO_OS_GPIO_HIGH;
+		break;
+	}
+
+	ret = no_os_gpio_set_value(desc->en0, en0);
+	if (ret)
+		return ret;
+
+	ret = no_os_gpio_set_value(desc->en1, en1);
```

Comment:
You can directly return here.
return no_os_gpio_set_value();

Reply:
This is noted.